src/components/picker/index.tsx

```diff
@@ -208,21 +203,21 @@ const Picker = React.forwardRef((props: PickerProps, ref) => {
         onCancel: cancelSelect
       };
 
-      return renderCustomModal(modalProps);
+      return componentProps?.customModal && componentProps.customModal?.(modalProps);
```

Comment:
Why checking again componentProps?.customModal? You have an if statement line 194

Reply:
So how about we pass 'onCancel' instead of the whole headerProps? This way you won't need to use the deprecated topBarProps in the hook. The component will pass the proper callback or undefined